### PR TITLE
Pin sympy version to <1.10

### PR DESCRIPTION
**Pull request by 95-martin-orion**
Part of #5058. The latest sympy release (v1.10) appears to have broken our mypy tests: #4936, #4964, #5030

Pinning until we can bring ourselves up to date (assuming that's what we want to do here)

### CODE REVIEW

**Review by vtomole — APPROVED** [review]
LGTM.

File: cirq-core/requirements.txt [hunk 1]
```diff
@@ -13,6 +13,6 @@ numpy~=1.16
 pandas
 sortedcontainers~=2.0
 scipy
-sympy
+sympy==1.9
```
Comment (dstrain115) [inline cirq-core/requirements.txt:16]: Should we link the bug to fix type errors here?

**Review by dstrain115 — APPROVED** [review]

**Comment by mpharrigan** [comment]
pleeeeeeease no `==`. If <1.10 fixes, then consider doing that

**Comment by mpharrigan** [comment]
any time you use `==` in `install_requires` with `pip`'s new dependency resolver it can easily cause unsolvable dependency conflicts. As always, I will point out that you're supposed to pin in `requirements.txt` for CI but relax in `install_requires` for user consumption, but they are linked in our build

File: cirq-core/requirements.txt (hunk 1 above)
Comment (95-martin-orion) [reply cirq-core/requirements.txt]: Done.

**Comment by 95-martin-orion** [comment]
Changed to `sympy<1.10`.

**Review by vtomole — APPROVED** [review]

**Pull request merged**